### DISCUSSION TITLE
Accept Invite with EnvKey Self-Hosted

### DISCUSSION THREAD

**Discussion by vinhdoanthe** [envkey/envkey#28] [Q&A] [answered]
Hi,

I've just installed EnvKey self-hosted, created Organization, and sent invite for my co-workers.
But when my co-workers choose to accept my invitation, they got error as image below:

![image](https://user-images.githubusercontent.com/4367924/192931857-ea55daa8-6e9f-44b6-abed-ef1aec5065df.png)

I guest the reason is EnvKey UI does not know URL for my self-hosted organization. But I don't know how to configure it.

Anyone here could help me?

**Comment by danenania** [comment]
Hi @vinhdoanthe, happy to help you get this straightened out. Could you please ask your co-worker to zip up any files in ~/.envkey/logs and either attach the zip file here or send it to support@envkey.com? That should help to shed light on what's going wrong.

**Reply by vinhdoanthe** [reply]
Hi @danenania , I attached log file here: 
[desktop-2022-09-29.log](https://github.com/envkey/envkey/files/9680381/desktop-2022-09-29.log)
We host EnvKey server at https://envkeys.teky.edu.vn.

**Reply by danenania** [reply]
Hi @vinhdoanthe could you also attach the log file starting with `core-` please?

**Comment by mihagrgicjelen** [comment, marked as answer]
Hi @danenania 

Im encountering the exact same problem as @vinhdoanthe described.

This is the **log from ~/.envkey/logs/core**-<date>.
(no additional log output is created when entering encryption and invite tokens and clicking the "next" button.)

{"ts":"2023-02-23T10:31:53.469Z","msg":"on:before-quit","currentAppVersion":"2.2.12","appWillAutoExit":false}
{"ts":"2023-02-23T10:31:53.469Z","msg":"Stopping check updates loop..."}
{"ts":"2023-02-23T10:31:53.470Z","msg":"stopping core process if it's running inline..."}
{"ts":"2023-02-23T10:31:53.472Z","msg":"core process wasn't running inline."}
{"ts":"2023-02-23T10:31:53.473Z","msg":"on:closed","currentAppVersion":"2.2.12"}
{"ts":"2023-02-23T10:31:55.707Z","msg":"logger initialized with name desktop"}
{"ts":"2023-02-23T10:31:55.825Z","msg":"on:ready","version":"2.2.12"}
{"ts":"2023-02-23T10:31:55.935Z","msg":"startCore","keepAlive":true,"inlineOnly":false}
{"ts":"2023-02-23T10:31:56.566Z","msg":"Core process status","alive":"2.2.10"}
{"ts":"2023-02-23T10:31:56.566Z","msg":"Core process is already running"}
{"ts":"2023-02-23T10:31:56.566Z","msg":"core is running","currentAppVersion":"2.2.12"}
{"ts":"2023-02-23T10:31:56.567Z","msg":"Fetching root-device-key from OS credential store..."}
{"ts":"2023-02-23T10:31:56.567Z","msg":"CLI inside `pkg`"}
{"ts":"2023-02-23T10:31:56.567Z","msg":"loading keytar from __non_webpack_require__('keytar')"}
{"ts":"2023-02-23T10:31:56.570Z","msg":"loaded keytar from __non_webpack_require__('keytar')"}
{"ts":"2023-02-23T10:31:56.618Z","msg":"Fetched root-device-key."}
{"ts":"2023-02-23T10:31:56.620Z","msg":"Setting User-Agent with OS keyring auth token"}
{"ts":"2023-02-23T10:31:56.632Z","msg":"init updates loop"}
{"ts":"2023-02-23T10:31:56.635Z","msg":"app update loop initialized"}
{"ts":"2023-02-23T10:31:56.636Z","msg":"autoUpdater:info","data":["Checking for update"]}
{"ts":"2023-02-23T10:31:57.395Z","msg":"autoUpdater:info","data":["Update for version 2.2.12 is not available (latest version: 2.2.12, downgrade is disallowed)."]}
{"ts":"2023-02-23T10:31:57.614Z","msg":"","currentVersion":"2.2.10","nextVersion":"2.2.10"}
{"ts":"2023-02-23T10:31:57.615Z","msg":"finished checking updates","hasDesktopUpgrade":false}
{"ts":"2023-02-23T10:31:57.615Z","msg":"","currentVersion":"2.2.10","nextVersion":"2.2.10"}`

Here is also the "developer tools - **console" log:**
root.tsx:252 2023-02-23T10:36:34.665Z dispatch: envkey/client/RESET_EXTERNAL_AUTH
root.tsx:265 2023-02-23T10:36:34.686Z core result: envkey/client/RESET_EXTERNAL_AUTH
root.tsx:278 2023-02-23T10:36:34.686Z envkey/client/RESET_EXTERNAL_AUTH 2 diffs to apply
root.tsx:286 2023-02-23T10:36:34.686Z applied patch
root.tsx:82 2023-02-23T10:36:34.687Z updating core state {bytes: 2923}
root.tsx:289 2023-02-23T10:36:34.689Z set new state
root.tsx:252 2023-02-23T10:36:34.698Z dispatch: envkey/client/LOAD_INVITE
reconnecting-websocket-mjs.js:454 RWS> message event
root.tsx:217 2023-02-23T10:36:34.729Z received 3 socket message diffs
root.tsx:222 2023-02-23T10:36:34.729Z applied socket message diffs
root.tsx:82 2023-02-23T10:36:34.729Z updating core state {bytes: 2649}
root.tsx:227 2023-02-23T10:36:34.730Z set new state
root.tsx:265 2023-02-23T10:36:34.763Z core result: envkey/client/LOAD_INVITE
root.tsx:278 2023-02-23T10:36:34.763Z envkey/client/LOAD_INVITE 4 diffs to apply
root.tsx:286 2023-02-23T10:36:34.763Z applied patch
root.tsx:82 2023-02-23T10:36:34.763Z updating core state {bytes: 2899}
root.tsx:289 2023-02-23T10:36:34.764Z set new state


Any help would be appreciated.

**Reply by danenania** [reply]
Hi @mihagrgicjelen the logs you pasted above look like the logs from the `desktop` log file rather than `core`. If you would like to just zip up all the logs files in ~/.envkey/logs and attach, that would help.

**Reply by mihagrgicjelen** [reply]
Thank you @danenania , my bad. Here is the zip! (removed)

**Reply by danenania** [reply]
Thanks @mihagrgicjelen! We're investigating

**Reply by danenania** [reply]
@mihagrgicjelen A likely cause of this could be that a `host` header is not present on requests. This is relied on to set the host URL that an invitation should be loaded from.

**Reply by mihagrgicjelen** [reply]
@danenania Correct, NGINX did not forward the headers. It works now!
Thank you for investigating quickly and providing assistance! Envkey is a superb product!

**Reply by danenania** [reply]
I'm glad to hear that @mihagrgicjelen and thanks for the kind words!
